feat(dev): add `scheduleBuildIfStale` method to DevEngine

sapphi-red opened the pull request:
Added `scheduleBuildIfStale` method to DevEngine to know whether the build is triggered.

refs https://github.com/vitejs/rolldown-vite/pull/401

sapphi-red commented:
* **#6087** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6087?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6087?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#6086** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6086?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=6087).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

sapphi-red commented on packages/rolldown/src/api/dev/dev-engine.ts:
```diff
@@ -63,6 +63,18 @@ export class DevEngine {
+  async scheduleBuildIfStale(): Promise<boolean> {
+    const scheduled = await this.#inner.scheduleBuildIfStale();
+    if (scheduled) {
+      scheduled.wait();
```
```suggestion
      // don't wait here as we want to return the result without waiting the actual build
      scheduled.wait().catch(() => {});
```

hyf0 reviewed (approved):
I leave comment here https://github.com/vitejs/rolldown-vite/pull/401/files#r2332202442. No matter if they overlap, we should do fast shipping and then polish them later.

hyf0 commented:
### Merge activity

* **Sep 9, 6:52 AM UTC**: The merge label 'graphite: merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) once it meets the requirements.
* **Sep 9, 6:52 AM UTC**: `hyf0` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).
* **Sep 9, 7:09 AM UTC**: The [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) removed this pull request due to **downstack failures on PR #[6086](https://app.graphite.dev/github/pr/rolldown/rolldown/6086)**.
* **Sep 9, 7:09 AM UTC**: The [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) removed this pull request due to **downstack failures on PR #[6086](https://app.graphite.dev/github/pr/rolldown/rolldown/6086)**.
* **Sep 9, 7:42 AM UTC**: The merge label 'graphite: merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) once it meets the requirements.
* **Sep 9, 7:57 AM UTC**: `hyf0` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).
* **Sep 9, 8:26 AM UTC**: The [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) couldn't merge this PR because **it was not satisfying all requirements** (Failed CI: 'browser-test (macos-latest) / WASI Test').

sapphi-red committed:
feat(dev): add `scheduleBuildIfStale` method to DevEngine (#6087)

Added `scheduleBuildIfStale` method to DevEngine to know whether the build is triggered.

refs https://github.com/vitejs/rolldown-vite/pull/401
